src/sentry/services/hybrid_cloud/organizationmember_mapping/impl.py

```diff
@@ -0,0 +1,64 @@
+# Please do not use
+#     from __future__ import annotations
+# in modules such as this one where hybrid cloud service classes and data models are
+# defined, because we want to reflect on type annotations and avoid forward references.
+
+from dataclasses import fields
+from typing import Optional
+
+from sentry.models.organizationmembermapping import OrganizationMemberMapping
+from sentry.services.hybrid_cloud.organizationmember_mapping import (
+    OrganizationMemberMappingService,
+    RpcOrganizationMemberMapping,
+)
+
+
+class DatabaseBackedOrganizationMemberMappingService(OrganizationMemberMappingService):
+    def create(
+        self,
+        *,
+        organization_id: int,
+        role: str,
+        user_id: Optional[int] = None,
+        email: Optional[str] = None,
+        inviter_id: Optional[int] = None,
+        invite_status: Optional[int] = None,
+        idempotency_key: Optional[str] = "",
+    ) -> RpcOrganizationMemberMapping:
+        assert user_id or email, "Must set either user or email"
+        if idempotency_key:
```

Comment:
I think it is sufficient to simply leave the `update_or_create` here, where we match on `organization_id` and `user_id`, and update all other columns?  But I'm curious how `email` is used in our invite flows as is.

Reply:
Oh right, so on creation, we, don't have a user_id right? 

I think we actually do not want a user_id parameter here at all.  `email_id` and `organization_id` is all that we need to accept?

Reply:
again, creating and updating org member mappings should always be a safe and idempotent operation anyway, so no need for this parameter